src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
I'd rather this function accept immutable references. Accepting mutable reference to metrics will make it difficult to update metrics after export is called. We'd like the user to be able to do that. It makes sense from an API standpoint as well, export should be called only once, with a fixed number of metrics to export, also we should make sure that export cannot be called repetitively on the same client.